python/cuspatial/cuspatial/tests/binpreds/test_equals_only_binops.py

```diff
@@ -0,0 +1,570 @@
+import cupy as cp
+import geopandas as gpd
+import pandas as pd
+import pytest
+from shapely.geometry import LineString, MultiPoint, Point, Polygon
+
+import cudf
+
+import cuspatial
+
+
+def test_from_points_xy_large():
+    points = cuspatial.GeoSeries(
+        cuspatial.core._column.geocolumn.GeoColumn._from_points_xy(
+            cudf.core.column.column.as_column(
+                cp.arange(10000000, dtype="float64")
+            )
+        )
+    )
+    assert points.geom_equals(points).all()
+
+
+def test_point_geom_equals_point():
+    gpdpoint1 = gpd.GeoSeries([Point(0, 0)])
+    gpdpoint2 = gpd.GeoSeries([Point(0, 0)])
+    point1 = cuspatial.from_geopandas(gpdpoint1)
+    point2 = cuspatial.from_geopandas(gpdpoint2)
+    got = point1.geom_equals(point2)
+    expected = gpdpoint1.geom_equals(gpdpoint2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+@pytest.mark.parametrize(
+    "lhs",
+    [
+        [Point(0, 0), Point(0, 0), Point(0, 0)],
+        [Point(1, 1), Point(1, 1), Point(1, 1)],
+        [Point(2, 2), Point(2, 2), Point(2, 2)],
+    ],
+)
+def test_3_points_equals_3_points_one_equal(lhs):
+    gpdpoint1 = gpd.GeoSeries(lhs)
+    gpdpoint2 = gpd.GeoSeries([Point(0, 0), Point(1, 1), Point(2, 2)])
+    point1 = cuspatial.from_geopandas(gpdpoint1)
+    point2 = cuspatial.from_geopandas(gpdpoint2)
+    got = point1.geom_equals(point2)
+    expected = gpdpoint1.geom_equals(gpdpoint2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_10000_points_geom_equals_10000_points(point_generator):
+    gpdpoints1 = gpd.GeoSeries([*point_generator(10000)])
+    gpdpoints2 = gpd.GeoSeries([*point_generator(10000)])
+    points1 = cuspatial.from_geopandas(gpdpoints1)
+    points2 = cuspatial.from_geopandas(gpdpoints2)
+    got = points1.geom_equals(points2)
+    expected = gpdpoints1.geom_equals(gpdpoints2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_linestring_geom_equals_linestring():
+    gpdline1 = gpd.GeoSeries([LineString([(0, 0), (1, 1)])])
+    gpdline2 = gpd.GeoSeries([LineString([(0, 0), (1, 1)])])
```

Comment:
What happens if
```suggestion
    gpdline1 = gpd.GeoSeries([LineString([(0, 0), (1, 1)])])
    gpdline2 = gpd.GeoSeries([LineString([(1, 1), (0, 0)])])
```
?

Reply:
Because this isn't the final iteration of `geom_equals`, I'm going to settle for the test case that I have so far, which I think covers most usage. As I refactor the design and work through the intersection cases I will create exhaustive tests for all combinations and types.